### ISSUE TITLE
CI/CD operation to run tests on every PR

### ISSUE THREAD

**Issue by AkMo3** [scorelab/LabelLab#627]
**Is your feature request related to a problem? Please describe.**
Currently, reviewers have to trust the contributors that the test present are passing or they have to clone the repo and test themselves. Contributors also usually do not have an idea on how the code reacts with different versions of Node and Python.

**Describe the solution you'd like**
By using Github Action, we can implement CI/CD tests that would run on all new PR, every subsequent push to a PR, as well as on the main repo. This would enforce strict quality control and reduce human error, which will be immensely helpful in the long term as we integrate more tests.

**Describe alternatives you've considered**
We can use CircleCI in place of Github Actions.

**Comment by AkMo3** [comment]
@amlannandy What do you think of it? Other contributors as well are welcome to present their views.

**Comment by amlannandy** [comment]
@AkMo3 Sounds good. This is something we have also been planning for a while.
But it's a pretty significant task, so it'd be better if we start work on this after GSoC coding period begins.
You can add it to your GSoC proposal as a milestone. But not something we can work on right now.

**Comment by AkMo3** [comment]
Sure!

**Comment by M-A-D-A-R-A** [comment]
> **Is your feature request related to a problem? Please describe.** Currently, reviewers have to trust the contributors that the test present are passing or they have to clone the repo and test themselves. Contributors also usually do not have an idea on how the code reacts with different versions of Node and Python.
> 
> **Describe the solution you'd like** By using Github Action, we can implement CI/CD tests that would run on all new PR, every subsequent push to a PR, as well as on the main repo. This would enforce strict quality control and reduce human error, which will be immensely helpful in the long term as we integrate more tests.
> 
> **Describe alternatives you've considered** We can use CircleCI in place of Github Actions.

Yeah that will help us and other contributors to make PRs better and in Flutter to make sure your application is delivered to testers and validated on a frequent basis without resorting to manual workflows, some of the manual

> [GitHub Actions](https://github.com/features/actions)
- Example: Flutter Gallery’s [Github Actions workflows](https://github.com/flutter/gallery/tree/master/.github/workflows)
- Example: [Github Action in Flutter Project](https://github.com/nabilnalakath/flutter-githubaction)
> [Cirrus](https://cirrus-ci.org/)

**Comment by M-A-D-A-R-A** [comment]
Should we set up some workflow action for the mobile application before we start the feature implementation  becoz we are just getting started in development and from now on each PR will be validated? @amlannandy your thoughts?

**Comment by amlannandy** [comment]
> Should we set up some workflow action for the mobile application before we start the feature implementation becoz we are just getting started in development and from now on each PR will be validated? @amlannandy your thoughts?

Sounds good. You can move forward with this.
